posthog/api/test/test_user.py

```diff
@@ -240,6 +240,44 @@ def test_update_current_user(self, mock_capture, mock_identify_task):
             },
         )
 
+    @patch("posthog.tasks.user_identify.identify_task")
+    @patch("posthoganalytics.capture")
+    def test_user_can_cancel_own_email_change_request(self, _mock_capture, _mock_identify_task):
+        self.user.pending_email = "another@email.com"
+        self.user.save()
+
+        response = self.client.patch("/api/users/cancel_email_change_request")
+
+        response_data = response.json()
+        assert response.status_code == status.HTTP_200_OK
+        assert response_data["pending_email"] is None
+
+    @patch("posthog.tasks.user_identify.identify_task")
+    @patch("posthoganalytics.capture")
+    def test_user_cannot_cancel_email_change_request_if_it_doesnt_exist(self, _mock_capture, _mock_identify_task):
+        # Fire a call to the endpoint without priming the User with a pending_email field
+
+        response = self.client.patch("/api/users/cancel_email_change_request")
+
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+    @patch("posthog.tasks.user_identify.identify_task")
+    @patch("posthoganalytics.capture")
+    def test_user_cannot_cancel_other_users_email_change_request(self, _mock_capture, _mock_identify_task):
+        another_user = User.objects.create(email="user@example.com", distinct_id=str(uuid.uuid4()))
+        another_team = Team.objects.create(name="Another team", organization=self.organization)
+
+        response = self.client.patch(
+            "/api/users/cancel_email_change_request",
+            # even if someone tries to send a different user or team they are ignored
+            {
```

Comment:
There's no body on the request so this shouldn't been needed. Since the user is retrieved from the user on the request, we shouldn't need to test this.